### PR TITLE
Align rule-splitters for Agents and the Public

### PR DESCRIPTION
**Pull request by Vinnl**
This is somewhat complex code that is mostly equivalent to both,
and will also be mostly equivalent for when we write support for
setting Group access and other potential future targets. Thus, to
prevent confusion, make sure that the functions actually _look_
mostly the same in addition to _behaving_ mostly the same.

# Checklist

- [ ] All acceptance criteria are met.
- [x] Relevant documentation, if any, has been written/updated.
- [x] The changelog has been updated, if applicable.
- [x] New functions/types have been exported in `index.ts`, if applicable.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

**Comment by NSeydoux** [comment]
Do you think it would be worth considering factoring these two (soon three) functions into one, or would it add an additional complexity greater that the gain in reducing redundancy ?

**Comment by Vinnl** [comment]
Yeah I was thinking about that too - in fact, not just these functions, but the entire files are mostly the same. But I'm pretty torn; it feels like it would be very hard to follow, and the function names would get even more unwieldy, probably. So the answer is: maybe, I don't know :shrug:
